x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/MultiRowTestCaseSupplier.java

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.test.ESTestCase.randomBoolean;
+import static org.elasticsearch.test.ESTestCase.randomList;
+import static org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier.TypedDataSupplier;
+
+/**
+ * Extension of {@link TestCaseSupplier} that provided multi-row test cases.
+ */
+public final class MultiRowTestCaseSupplier {
+
+    private MultiRowTestCaseSupplier() {}
+
+    public static List<TypedDataSupplier> multiRowIntCases(int minRows, int maxRows, int min, int max, boolean includeZero) {
+        List<TypedDataSupplier> cases = new ArrayList<>();
+
+        if (0 <= max && 0 >= min && includeZero) {
+            cases.add(
+                new TypedDataSupplier("<0 multi-row int>", () -> randomList(minRows, maxRows, () -> 0), DataType.INTEGER, false, true)
+            );
+        }
+
+        if (max != 0) {
+            cases.add(
+                new TypedDataSupplier(
+                    "<" + max + " multi-row int>",
+                    () -> randomList(minRows, maxRows, () -> max),
+                    DataType.INTEGER,
+                    false,
+                    true
+                )
+            );
+        }
+
+        if (min != 0 && min != max) {
+            cases.add(
+                new TypedDataSupplier(
+                    "<" + min + " multi-row int>",
+                    () -> randomList(minRows, maxRows, () -> min),
+                    DataType.INTEGER,
+                    false,
+                    true
+                )
+            );
+        }
+
+        int lower = Math.max(min, 1);
+        int upper = Math.min(max, Integer.MAX_VALUE);
+        if (lower < upper) {
+            cases.add(
+                new TypedDataSupplier(
+                    "<positive multi-row int>",
```

Comment:
What about `<positive ints>`? I'm mostly nit picking to try and save screen real estate when I mouse over tests.